app/src/main/java/org/xbmc/kore/ui/sections/remote/PlaylistFragment.java

```diff
@@ -289,101 +344,121 @@ public void playerNoResultsYet() {
+        // BUG: When playing movies playlist stops, audio tab gets selected when it contains a playlist.
+        // We might want a separate var to check if something has already played and turn off automatic
+        // playlist switching if playback stops
+        if (playerState == PLAYER_STATE.STOPPED && lastGetActivePlayerResult == null && !userSelectedTab) { // do not automatically switch to first available playlist if user manually selected a playlist
+            playlistsBar.selectTab(playlists.get(0).type);
+        }
+        displayPlaylist();
+    }
+
+    @Override
+    public void playlistOnError(int errorCode, String description) {
+        displayErrorGettingPlaylistMessage(description);
+    }
+
+    private void updatePlaylists(ArrayList<GetPlaylist.GetPlaylistResult> playlists) {
+        for (GetPlaylist.GetPlaylistResult getPlaylistResult : playlists) {
+            playlistsBar.setHasPlaylistAvailable(getPlaylistResult.type, true);
+
+            PlaylistHolder playlistHolder = this.playlists.get(getPlaylistResult.type);
+
+            if (playlistHolder == null) {
+                playlistHolder = new PlaylistHolder();
+                this.playlists.put(getPlaylistResult.type, playlistHolder);
+            }
+
+            playlistHolder.setPlaylist(getPlaylistResult);
         }
     }
 
-    private void displayPlaylist(final ListType.ItemsAll getItemResult,
-                                 final List<ListType.ItemsAll> playlistItems) {
-        if (playlistItems.isEmpty()) {
+    private void displayPlaylist() {
+        switchToPanel(R.id.playlist);
+
+        PlaylistHolder playlistHolder = playlists.get(playlistsBar.getSelectedPlaylistType());
+        if (playlistHolder == null) {
             displayEmptyPlaylistMessage();
             return;
         }
-        switchToPanel(R.id.playlist);
+
+        GetPlaylist.GetPlaylistResult getPlaylistResult = playlistHolder.getPlaylistResult;
+        if ( getPlaylistResult == null ) {
+            displayEmptyPlaylistMessage();
+            return;
+        }
+
+        // JSON RPC does not support picture items in Playlist.Item so we disable item movement
+        // for the picture playlist
+        if (getPlaylistResult.type.contentEquals(ListType.ItemBase.TYPE_PICTURE))
+            playlistListView.enableItemDragging(false);
+        else
+            playlistListView.enableItemDragging(true);
 
         //If a user is dragging a list item we must not modify the adapter to prevent
         //the dragged item's adapter position from diverging from its listview position
-        if (!playlistListView.isItemBeingDragged()) {
-            // Set items, which call notifyDataSetChanged
-            playListAdapter.setPlaylistItems(playlistItems);
-            highlightItem(getItemResult, playlistItems);
-        } else {
-            highlightItem(getItemResult, playListAdapter.playlistItems);
+        if ( ! playlistListView.isItemBeingDragged() ) {
```

Comment:
For consistency with the rest of the code, no spaces inside ()
And i would prefer no spaces after ! also